lib/internal/stream_base_commons.js

```diff
@@ -178,6 +189,38 @@ function onStreamRead(arrayBuffer) {
   }
 }
 
+function setStreamTimeout(msecs, callback) {
+  if (this.destroyed)
+    return;
+
+  this.timeout = msecs;
+
+  // Type checking identical to timers.enroll()
+  msecs = validateTimerDuration(msecs);
+
+  // Attempt to clear an existing timer in both cases -
+  //  even if it will be rescheduled we don't want to leak an existing timer.
+  clearTimeout(this[kTimeout]);
+
+  if (msecs === 0) {
+    if (callback !== undefined) {
+      if (typeof callback !== 'function')
+        throw new ERR_INVALID_CALLBACK();
```

Comment:
I think this makes this change semver-major because of the net changes. Can you also add a test for net that catches this error?